internal/controller/nimbuild_controller.go

```diff
@@ -0,0 +1,956 @@
+	if nimCache.Spec.Source.NGC == nil {
+		return nil
+	}
+
+	existingConfig := &corev1.ConfigMap{}
+	cmName := getManifestConfigName(nimCache)
+	err := r.Get(ctx, client.ObjectKey{Name: cmName, Namespace: nimBuild.Namespace}, existingConfig)
+	if err != nil {
+		logger.Error(err, "failed to get configmap of the local model manifest", "name", cmName)
+		return err
+	}
+
+	// Create a temporary pod for parsing local model manifest
+	pod := constructLocalManifestPodSpec(nimCache, nimBuild, r.orchestratorType)
+	// Add nimBuild as owner for watching on status change
+	if err := controllerutil.SetControllerReference(nimBuild, pod, r.GetScheme()); err != nil {
+		return err
+	}
+	err = r.createPod(ctx, pod)
+	if err != nil {
+		logger.Error(err, "failed to create", "pod", pod.Name)
+		return err
+	}
+
+	existingPod := &corev1.Pod{}
+	err = r.Get(ctx, client.ObjectKey{Name: pod.Name, Namespace: nimBuild.Namespace}, existingPod)
+	if err != nil {
+		logger.Error(err, "failed to get pod for model selection", "pod", pod.Name)
+		return err
+	}
+
+	if existingPod.Status.Phase != corev1.PodRunning {
+		// requeue request with delay until the pod is ready
+		return nil
+	}
+
+	// Extract manifest file
+	output, err := k8sutil.GetPodLogs(ctx, existingPod, NIMBuildManifestContainerName)
+	if err != nil {
+		logger.Error(err, "failed to get pod logs for parsing model manifest file", "pod", pod.Name)
+		return err
+	}
+
+	if output == "" {
+		logger.Info("Requeuing to wait for the manifest to be copied from the container")
+		return nil
+	}
+
+	parser := nimparserutils.GetNIMParser([]byte(output))
+	// Parse the file
+	manifest, err := parser.ParseModelManifestFromRawOutput([]byte(output))
+	if err != nil {
+		logger.Error(err, "Failed to parse model manifest from the pod")
+		return err
+	}
+	logger.V(2).Info("local manifest file", "nimbuild", nimBuild.Name, "manifest", manifest)
+
+	// Create a ConfigMap with the model manifest file for re-use
+	err = r.updateManifestConfigMap(ctx, nimCache, &manifest)
+	if err != nil {
+		logger.Error(err, "Failed to create model manifest config map")
+		return err
+	}
+
+	// Model manifest is successfully extracted, cleanup temporary pod
+	err = r.Delete(ctx, existingPod)
+	if err != nil && !errors.IsNotFound(err) {
+		logger.Error(err, "failed to delete", "pod", pod.Name)
+		// requeue request with delay until the pod is cleaned up
+		// this is required as NIM containers are resource heavy
+		return err
+	}
+
+	// To Do: Explore changing the profile list to a map for faster lookups
+	for _, profileName := range manifest.GetProfilesList() {
+		for _, selectedProfile := range nimCache.Status.Profiles {
+			if selectedProfile.Name == profileName {
+				break
+			}
+		}
+		logger.Info("Adding profile to NIMCache status", "profileName", profileName)
+		// If the profile is not found, add it to the status
+		nimCache.Status.Profiles = append(nimCache.Status.Profiles, appsv1alpha1.NIMProfile{
```

Comment:
Looks like we end up adding duplicate profiles here. If at the end of build process, a new profile is created, then we should relate that to the original profile ID selected (or user provided). Only that new profile ID should be added in the NIMCache status with the build tags.